### DISCUSSION TITLE
How to add green tick to digital signature

### DISCUSSION THREAD

**Discussion by ankitkr18218** [MatthiasValvekens/pyHanko#282] [Q&A]
Can't find way to add green tick to digital signature which should appear on validating the signature in pyhanko, could someone help.

**Comment by MatthiasValvekens** [comment]
Hi @ankitkr18218,


This comes up every so often. It's a proprietary Acrobat feature from back in the days of yore, was deprecated very shortly after it was introduced, and was never part of ISO-standard PDF.

It was deprecated for very good reasons: the validation status of a signature should _never_ be part of its visual appearance on the page, because the appearance on the page is trivial to spoof. As such, users should not rely on any in-band visual cues to judge the validity of a signature. Rather, they should use the signature pane in their preferred viewer (or use external validation tooling).

Long story short: this is bad security practice and therefore will never be "officially" supported in pyHanko. For what it's worth, pyHanko is in good company here---most vendors I know of stopped supporting this a long time ago, and even Adobe all but killed the feature completely (except in some very specific legacy corner cases).

If you are still convinced that you somehow need this feature, I suggest looking up "signature appearance layers" in one of the PDF specs from the early-to-mid '00s and trying to emulate that in pyHanko. ;)
